crates/sui-types/src/gas.rs

```diff
@@ -27,6 +28,53 @@ use sui_cost_tables::{
 };
 use sui_protocol_config::*;
 
+// A bucket defines a range of units that will be priced the same.
+// A cost for the bucket is defined to make the step function non linear.
+#[allow(dead_code)]
+struct ComputationBucket {
+    min: u64,
+    max: u64,
+    cost: u64,
+}
+
+impl ComputationBucket {
+    fn new(min: u64, max: u64, cost: u64) -> Self {
+        ComputationBucket { min, max, cost }
+    }
+
+    fn simple(min: u64, max: u64) -> Self {
+        ComputationBucket {
+            min,
+            max,
+            cost: max,
+        }
+    }
+}
+
+fn get_bucket_cost(table: &[ComputationBucket], computation_cost: u64) -> u64 {
+    for bucket in table {
+        if bucket.max >= computation_cost {
+            return bucket.cost;
+        }
+    }
+    MAX_BUCKET_COST
+}
+
+// for a RPG of 1000 this amounts to about 1 SUI
+const MAX_BUCKET_COST: u64 = 1_000_000;
+
+// define the bucket table for computation charging
+static COMPUTATION_BUCKETS: Lazy<Vec<ComputationBucket>> = Lazy::new(|| {
```

Comment:
Just a note for a future: we can probably optimize this with a segment tree + range queries. But it won't change the semantics, the number of tiers is low, and we can change it easily later. So this if fine for now :)